frontend/src/bundles/common/components/sidebar/hooks/use-collapse/use-collapse.hook.ts

```diff
@@ -0,0 +1,46 @@
+import { actions as authActions } from '~/bundles/auth/store/auth.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+    useEffect,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { MD } from './constants/constants.js';
+
+const useCollapse = (): {
+    isCollapsed: boolean;
+    setToggle: () => void;
+} => {
+    const isSidebarCollapsed = useAppSelector(
+        ({ auth }) => auth.isSidebarCollapsed,
+    );
+    const dispatch = useAppDispatch();
+
+    const [isCollapsed, setIsCollapsed] = useState<boolean>(isSidebarCollapsed);
```

Comment:
I think you can remove local state and use only values from store. Because with current implementation you have the same data in two different places and therefore when updating data you need to change it in two different places which is redundant.

Reply:
With that new implementation you can dispatch `toggle` action inside your `handleToggle` handler and resize logic can go into the component so there will be no need for custom hook.